### PR TITLE
AKU-445, AKU-446: Document Library upload scoping issues

### PR DESCRIPTION
**Pull request by draperd**
This addresses both https://issues.alfresco.com/jira/browse/AKU-445 and https://issues.alfresco.com/jira/browse/AKU-446 to resolve issues preventing uploading and refreshing Document Libraries when those Document Libraries have been given a specific pubSubScope.

### CODE REVIEW

File: aikau/src/main/resources/alfresco/documentlibrary/_AlfCreateContentMenuItemMixin.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+       * @type {object[]}
+       * @default null
+       */
+      widgets: null,
+
+      /**
+       * 
+       * @instance
+       */
+      onClick: function alfresco_documentlibrary__AlfCreateContentPermissionsMixin__onClick(/*jshint unused:false*/ evt) {
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/documentlibrary/_AlfCreateContentMenuItemMixin.js]: Incorrect function name for module `alfresco/documentlibrary/_AlfCreateContentMenuItemMixin`

File: aikau/src/main/resources/alfresco/documentlibrary/_AlfCreateContentMenuItemMixin.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+      /**
+       * This handles publications on the 
+       * [metadataChangeTopic]{@link module:alfresco/documentlibrary/_AlfDocumentListTopicMixin#metadataChangeTopic}
+       * topic to set the location of the current folder 
+       * 
+       * @instance
+       * @param {object} payload 
+       */
+      onCurrentNodeChange: function alfresco_documentlibrary__AlfCreateContentPermissionsMixin__onCurrentNodeChange(payload) {
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/documentlibrary/_AlfCreateContentMenuItemMixin.js]: Incorrect function name for module `alfresco/documentlibrary/_AlfCreateContentMenuItemMixin`

File: aikau/src/main/resources/alfresco/menus/AlfFilteringMenuItem.js [hunk 3]
```diff
@@ -44,9 +44,9 @@ define(["dojo/_base/declare",
        * @instance
        */
       postCreate: function alfresco_menus_AlfMenuItem__postCreate() {
-         if (this.filterTopic != null)
+         if (this.filterTopic)
          {
-            this.alfSubscribe(this.filterTopic, lang.hitch(this, "filter"));
+            this.alfSubscribe(this.filterTopic, lang.hitch(this, this.filter));
          }
          this.inherited(arguments);
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/menus/AlfFilteringMenuItem.js]: ___[Comment only]___ It'd be more concise to just have:
`this.filterTopic && this.alfSubscribe(this.filterTopic, lang.hitch(this, this.filter));`

File: aikau/src/main/resources/alfresco/renderers/Thumbnail.js [hunk 4]
```diff
@@ -393,6 +393,7 @@ define(["dojo/_base/declare",
          {
             this.addUploadDragAndDrop(this.imgNode);
             this.addNodeDropTarget(this.imgNode);
+            this._currentNode = this.currentItem.node;
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/renderers/Thumbnail.js]: `this.currentItem` definitely present?

File: aikau/src/main/resources/alfresco/services/UploadService.js [hunk 5]
```diff
@@ -232,6 +232,7 @@ define(["dojo/_base/declare",
             // Store a response topic to publish on when the upload is complete...
             // This allows for scoped responses to be processed without pollution...
             this.currentResponseTopic = payload.alfResponseTopic;
+            this.currentResponseScope = payload.alfResponseScope;
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/services/UploadService.js]: `this.currentResponseScope = payload.responseScope || payload.alfOriginScope;` ???

File: aikau/src/main/resources/alfresco/menus/AlfFilteringMenuItem.js (hunk 3 above)
Comment (davidcognite) [inline aikau/src/main/resources/alfresco/menus/AlfFilteringMenuItem.js]: Concise is not necessarily better. Using an if block makes the code more readable.

**Pull request merged**